nengo/utils/threading.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import
+
+import collections
+import threading
+
+
+class ThreadLocalContext(threading.local, collections.Sequence):
```

Comment:
This could be used for things other than our Network context, right? Should it have a more general name, like `ThreadLocalList`? And call the internal variable `_list` or something?